segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -604,6 +761,12 @@ private Void handleException(long requestId, String segment, String operation, T
         } else if (u instanceof BadOffsetException) {
             BadOffsetException badOffset = (BadOffsetException) u;
             invokeSafely(connection::send, new SegmentIsTruncated(requestId, segment, badOffset.getExpectedOffset(), clientReplyStackTrace), failureHandler);
+        } else if (u instanceof TableSegmentNotEmptyException) {
+            log.warn(requestId, "Table segment '{}' is not empty to perform '{}'.", segment, operation);
+            invokeSafely(connection::send, new TableSegmentNotEmpty(requestId, segment, clientReplyStackTrace), failureHandler);
+        } else if (u instanceof ConditionalTableUpdateException) {
```

Comment:
I believe @shiveshr may want to distinguish between `BadKeyVersionException` and `KeyNotExistsException` (both of which derive from `ConditionalTableUpdateException`. I suggest we add special handlers for those too, above this one (and keep `ConditionalTableUpdateException`.

Reply:
Created `WireCommands.TableKeyDoesNotExist` and `WireCommands.TableKeyBadVersion`.